src/Octokit.Webhooks.AzureFunctions/GitHubWebhooksHttpFunction.cs

```diff
@@ -0,0 +1,150 @@
+namespace Octokit.Webhooks.AzureFunctions;
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Mime;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+
+/// <summary>
+/// A class containing an Azure Function that processes GitHub webhooks.
+/// </summary>
+public partial class GitHubWebhooksHttpFunction
+{
+    private readonly IOptions<GitHubWebhooksOptions> options;
+
+    public GitHubWebhooksHttpFunction(IOptions<GitHubWebhooksOptions> options) => this.options = options;
+
+    [Function(nameof(MapGitHubWebhooksAsync))]
+    public async Task<HttpResponseData?> MapGitHubWebhooksAsync(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "POST", Route = "github/webhooks")] HttpRequestData req,
+        FunctionContext ctx)
+    {
+        var logger = ctx.InstanceServices.GetService<ILogger<GitHubWebhooksHttpFunction>>()!;
+
+        // Verify content type
+        if (!VerifyContentType(req, MediaTypeNames.Application.Json))
+        {
+            Log.IncorrectContentType(logger);
+            return req.CreateResponse(HttpStatusCode.BadRequest);
+        }
+
+        // Get body
+        var body = await GetBodyAsync(req).ConfigureAwait(false);
+
+        // Verify signature
+        if (!VerifySignature(req, this.options.Value.Secret, body))
+        {
+            Log.SignatureValidationFailed(logger);
+            return req.CreateResponse(HttpStatusCode.BadRequest);
+        }
+
+        // Process body
+        try
+        {
+            var service = ctx.InstanceServices.GetService<WebhookEventProcessor>()!;
```

Comment:
If you use the required variant then the nullable annotation should be that it won't ever return null as it would throw if the service wasn't registered instead.

```suggestion
            var service = ctx.InstanceServices.GetRequiredService<WebhookEventProcessor>();
```